### ISSUE TITLE
Fix color contrast in search results

### ISSUE THREAD

**Issue by atsansone** [flutter/website#10538]
From https://buganizer.corp.google.com/issues/337302544

### Observed Results

The results links and code numbers do not have enough contrast ratio.

Color contrast of 3.74 (foreground color: #009933, background color: #ffffff, font size: 9.8pt (13px), font weight: normal).

Element has insufficient color contrast of 1.54 (foreground color: #d0d0d2, background color: #ffffff, font size: 10.5pt (14px), font weight: normal).

* [Screenshot](https://screenshot.googleplex.com/APn7GTggPHzaDnd.png)
* [Screenshot](https://screenshot.googleplex.com/9GZwEmjiqqmBvuj.png)

### Expected Results

Ensure that the color contrast for the links should be at least 4.5:1 to be accessible to everyone.

### User Impact

Low vision users are affected because they cannot distinguish those links.

### Steps to Reproduce

1. Enabled the color contrast companion tool.
1. Open the following pages:
   * https://docs.flutter.dev/search?q=devtools
   * https://docs.flutter.dev/cookbook/design/snackbars#interactive-example
1. Continue the linear navigation using the <kbd>Tab</kbd> key.
1. Observe that the results links colors and code numbers.

### GAR Assessment

[Flutter Docs](https://a11ytracker.corp.google.com/products/c2fdcf18-53d2-475d-b8b0-e5dedf8cb900?e=ProductLabels)

### GAR Details

[1.14. Is there sufficient color contrast between the text and the background?](go/GAR-web#color-contrast)

### Environment Details

Machine: MAC/ChromeBook
Browser: Chrome Version 120.0.6099.272 (Official Build) (64-bit)
Screen reader: NA.
Bug filling time: 15 mins.
